packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -468,19 +468,68 @@ export class SDKClient {
     const transactionType = transaction.constructor.name;
     const requestIdPrefix = formatRequestIdMessage(requestId);
     const currentDateNow = Date.now();
+    let gasUsed: any = 0;
+    let transactionFee: number = 0;
+    let transactionId: string = '';
     try {
-      const shouldLimit = this.hbarLimiter.shouldLimit(currentDateNow, SDKClient.transactionMode, callerName);
-      if (shouldLimit) {
+      if (this.hbarLimiter.shouldLimit(currentDateNow, SDKClient.transactionMode, callerName))
         throw predefined.HBAR_RATE_LIMIT_EXCEEDED;
-      }
 
       this.logger.info(`${requestIdPrefix} Execute ${transactionType} transaction`);
-      const resp = await transaction.execute(this.clientMain);
+      const transactionResponse = await transaction.execute(this.clientMain);
+
+      try {
+        const transactionRecord = await transactionResponse.getRecord(this.clientMain);
+        transactionId = transactionRecord.transactionId.toString();
+
+        // get transactionFee and gasUsed for metrics
+        /**
+         * @todo: Determine how to separate the fee charged exclusively by the operator because
+         *        the transactionFee below includes the entire charges of the transaction,
+         *        with some portions paid by tx.from, not the operator.
+         */
+        transactionFee = transactionRecord.transactionFee.toTinybars().toNumber();
+        gasUsed = transactionRecord?.contractFunctionResult?.gasUsed.toNumber();
+      } catch (e: any) {
```

Comment:
Q: why the catch?
There's already a larger catch that should handle any transaction execution failures

Reply:
updated with new changes.